### PR TITLE
feat(IGDB): add presence

### PR DESCRIPTION
**Pull request by kyrie25**
#### Resolves discussion #4449 

<details>
<summary> Proof of the presence working </summary>
<br>

image
image
image
image
image

</details>

### CODE REVIEW

**Review by EncryptedDev — DISMISSED** [review]

File: websites/I/IGDB/presence.ts [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+  } else if (path.startsWith("/franchises")) {
+    path === "/franchises"
+      ? (presenceData.details = "Browsing all franchises")
+      : ((presenceData.details = "Browsing a franchise"),
+        (presenceData.state = document.title));
+  } else if (path.startsWith("/categories")) {
+    path === "/categories"
+      ? (presenceData.details = "Browsing all categories")
+      : ((presenceData.details = "Browsing a category"),
+        (presenceData.state = document.title.slice(0, -5)));
```
Comment (Bas950) [inline websites/I/IGDB/presence.ts:77]: plz no, dont do it like this. just make it in if case and not all these (), (), ();

**Review by Bas950 — DISMISSED** [review]

File: websites/I/IGDB/presence.ts (hunk 1 above)
Comment (kyrie25) [reply websites/I/IGDB/presence.ts:77]: It can't be that bad right lol, I'll rewrite the code when I can

Comment (kyrie25) [reply websites/I/IGDB/presence.ts]: done

File: websites/I/IGDB/presence.ts [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+        "#content-page > div.loaded > div > div.gamepage-header-info > div.gamepage-cover > img"
+      ).src;
+      presenceData.smallImageKey = "logo";
+      presenceData.smallImageText = "IGDB";
+    }
+  } else if (path.startsWith("/genres")) {
+    if (path === "/genres") presenceData.details = "Browsing all genres";
+    else {
+      (presenceData.details = "Browsing a genre"),
+        (presenceData.state = document.title.slice(0, -5));
```
Comment (EncryptedDev) [inline websites/I/IGDB/presence.ts:36]: Brackets aren't needed?
```suggestion
      presenceData.details = "Browsing a genre",
      presenceData.state = document.title.slice(0, -5);
```

File: websites/I/IGDB/presence.ts [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+    if (path === "/platforms") presenceData.details = "Browsing all platforms";
+    else {
+      delete presenceData.largeImageKey,
+        (presenceData.details = "Viewing a platform"),
+        (presenceData.state = document.title),
+        (presenceData.largeImageKey = document.querySelector<HTMLImageElement>(
+          "#content-page > div > div > div.row > div.col-sm-4 > img"
+        ).src),
+        (presenceData.smallImageKey = "logo"),
+        (presenceData.smallImageText = "IGDB");
```
Comment (EncryptedDev) [inline websites/I/IGDB/presence.ts:48]: Fix brackets and replace the `,` for `;`

**Review by EncryptedDev — CHANGES_REQUESTED** [review]

File: websites/I/IGDB/presence.ts (hunk 3 above)
Comment (kyrie25) [reply websites/I/IGDB/presence.ts:48]: Yeah I must have missed that part sorry

**Review by EncryptedDev — APPROVED** [review]
lgtm

**Review by Slowlife01 — APPROVED** [review]

**Pull request merged**